### PR TITLE
feat: Added LinearLRScheduler as a new component

### PR DESCRIPTION
**Pull request by fromm-m**
# What does this PR do?

This PR implementes the LinearLR sheduler (https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.LinearLR.html)

## General Changes
* ..

## Breaking Changes
* .. 

## Checklist before submitting final PR
- [x] My PR is minimal and addresses one issue in isolation
- [x] I have merged the latest version of the target branch into this feature branch
- [x] I have reviewed my own code w.r.t. correct implementation, missing type hints, proper documentation, etc.
- [x] I have run a sample config for model training
- [x] I have checked that all tests run through (`python tests/tests.py`)
- [ ] I have updated the internal changelog (`CHANGELOG_DEV.md`)

### CODE REVIEW

**Review by le1nux — APPROVED** [review]
LGTM! :-)

File: src/modalities/config/config.py [hunk 1]
```diff
@@ -185,6 +185,15 @@ class ConstantLRSchedulerConfig(BaseModel):
     verbose: bool = False
 
 
+class LinearLRSchedulerConfig(BaseModel):
+    optimizer: PydanticOptimizerIFType
+    start_factor: Annotated[float, Field(strict=True, ge=0.0, le=1.0)]
```
Comment (mali-git) [inline src/modalities/config/config.py:190]:
```suggestion
    start_factor: Annotated[float, Field(strict=True, gt=0.0, le=1.0)]
```

Comment (mali-git) [reply src/modalities/config/config.py:190]: To avoid a starting learning rate of 0.

**Review by mali-git — APPROVED** [review]
Minor change.
